### PR TITLE
Replace Telegraf agent with OpenTelemetry Collector

### PR DESCRIPTION
**Pull request by sky333999**
# Description of the issue
Please see <update_with_link_to_public_blog> for details.

# Description of changes
Please review individual commits for detailed description of changes.
A high level summary:
* Replace the telegraf agent with the open telemetry (OTel) collector. Fully backwards compatible.
* Introduce support for traces.
* CWA is now capable of operating with both telegraf & OTel receivers.
* Introduce an OTel adapter for the conversion of telegraf data model to OTel data model which enables telegraf input plugins to still be used.
* Leverage components from [OTel contrib](https://github.com/amazon-contributing/opentelemetry-collector-contrib) to replace local code for Container Insights & EMF pipelines.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Several test suites updated accordingly.

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml
Comment (sky333999) [inline .github/workflows/codeql-analysis.yml:1]: Should I add this back @SaxyPandaBear

**Review by straussb — APPROVED** [review]

**Review by SaxyPandaBear — APPROVED** [review]

File: .github/workflows/test-build.yml [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+        run: make amazon-cloudwatch-agent-linux amazon-cloudwatch-agent-windows package-rpm package-deb package-win
+
+      - name: Sign Build Files
+        if: contains(inputs.BucketKey, 'test') == false || steps.cached_binaries.outputs.cache-hit == false
+        run: for f in $(find build/bin/); do if [ ! -d $f ]; then echo "Signing file $f" && gpg --detach-sign $f ; fi ; done
+
+
+      - name: Upload to s3
+        if: contains(inputs.BucketKey, 'test') == false || steps.cached_binaries.outputs.cache-hit == false
+        # Copy the RPM to .../amazon_linux/... because BETA customers expect it there.
```
Comment (jefchien) [inline .github/workflows/test-build.yml:113]: nit: Remove comment.

File: internal/publisher/publisher_test.go [hunk 2]
```diff
@@ -61,51 +63,15 @@ func TestPublisher_PublishWithNonBlockFifoQueueSleep(t *testing.T) {
...
-	time.Sleep(3 * time.Second)
-	publisher.Close()
-	assert.Equal(t, []string{"req1", "req2", "req3", "req4"}, c.getResult())
-}
-
 func TestPublisher_DrainTimeout(t *testing.T) {
 	start := time.Now()
 	c := &testClient{}
-	publisher, _ := NewPublisher(NewBlockingFifoQueue(2), 1, 2*time.Second, c.publishWith5sLatency)
+	publisher, _ := NewPublisher(NewNonBlockingFifoQueue(2), 1, 2*time.Second, c.publishWith5sLatency)
```
Comment (jefchien) [inline internal/publisher/publisher_test.go:69]: So does the queue not affect the behavior? I'm guessing not since the unit tests pass.

**Review by jefchien — APPROVED** [review]

File: internal/publisher/publisher_test.go (hunk 2 above)
Comment (sky333999) [reply internal/publisher/publisher_test.go:69]: Yeah.. and since its only for the scope of this test, dont think its too concerning.

**Pull request merged**